### DISCUSSION TITLE
Support for closed loop stepper

### DISCUSSION THREAD

**Discussion by philGrz** [grblHAL/core#85] [Ideas]
is there already a support for closed loop steppers that can report step loss?

For this it the usage of the CS signal of SPI from the stepper modules could be used.

I want to implement this on a new board with RJ45 connectors
![StepperDriverInterface](https://user-images.githubusercontent.com/94231529/141648108-237e2c16-5dfb-42ed-9a04-29ac3e81c617.PNG)

**Comment by terjeio** [comment]
There is [support in the core](https://github.com/grblHAL/core/search?q=motor_fault), driver code has to be added to handle the pins.

**Reply by andrewmarles** [reply]
I tested this on the GRBLHAL2000 (IMX driver) which has hardware options to route the error signal from external closed loop stepper or servo drives.  It worked fine with the appropriate error shown on the console and I will see if I can dig up my build and make a PR for IMX.  In the end I found it was sufficient to just route the signal (hardware option) to the e-stop pin since that is the action that I'd take on motor error regardless, and doing it this way saves a GPIO pin dedicated to motor error.

**Comment by philGrz** [comment]
great, thx for information
